src/guides/components/styling.md

```diff
@@ -0,0 +1,142 @@
+# Using the Component Library
+
+Plugin pages provide a graphical user interface in the Caido application and can be customized.
+
+When running the `pnpm create @caido-community/plugin` command to initialize a new plugin project, you are given the option to use the [Vue.js](https://vuejs.org/) framework.
+
+When using this option, instead of building the plugin page in the frontend `index.ts` file, now the `App.vue` file is responsible for the plugin page's layout, state, and user interactions.
+
+However, the initialization, configuration, SDK setup, routing, and registration are still handled by `index.ts`.
+
+::: info
+For conceptual information regarding this guide - click [here](/concepts/frontend/ui.md).
+:::
+
+## Using Components
+
+To view all the components and their configuration options offered by the PrimeVue package - click [here](https://tailwind.primevue.org/vite/). You can also view them in the `~\node_modules\.pnpm\primevue@4.1.0_vue@X.X.XX_typescript@X.X.X_` directory of your project.
+
+::: warning
+Caido supports PrimeVue v4.1.0. If a style is not working properly, first try downgrading the `primevue` entry in the frontend `package.json` file. If this does not resolve the issue, please [let us know](https://docs.caido.io/report_bug.html)!
+:::
+
+### /packages/frontend/src/views/App.vue
```

Comment:
I think there's a lot of unnecessary code in this example.

We should simplify this snippet to only show stuff related to Primevue.

Let's remove all the unnecessary divs + `useSDK()` stuff. The `onGenerateClick()` could increment a number instead of calling a backend endpoint.